### DISCUSSION TITLE
Fate of xlink, opinion needed

### DISCUSSION THREAD

**Discussion by baybal** [edent/SuperTinyIcons#549] [General]
XML preamble `xlink:` and namespace declaration `xmlns:xlink="http://www.w3.org/1999/xlink" `is no longer needed in every web renderer these days. https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/href

This has been a proposal since SVG 1.2, and this is why some renders not explicitly supporting 2.0 has its support.

Instead of `<use x="-19" y="170" xlink:href="#o"/>` you can simply use `<use x="-19" y="170" href="#o"/>`

Support in browsers is certainly there, but I can't say so for offline renderers. Librsvg doesn't seem to do it despite having most complete 1.2 support of all offline renderers.

**Comment by edent** [comment]
Hmmm.... It looks like it still doesn't work in Safari. And Ubuntu doesn't seem to natively support it - either in EOM or file thumbnails.

I think we need to bring it back, for better compatibility. Or will that take any tiny SVG over the limit?

![twil](https://user-images.githubusercontent.com/837136/54318238-8e5c1880-45dd-11e9-9347-37357389e3f2.png)

**Comment by baybal** [comment]
Most Linux viewers use Librsvg. The question is if we continue this as a web first iconset.

Question about Safari, can you find somebody with a macintosh computer to check how things are in real life with last Safari version?

**Comment by edent** [comment]
Safari Version 12.0.3 (13606.4.5.3.1) - these icons are broken
![Screen Shot 2019-03-14 at 14 16 14](https://user-images.githubusercontent.com/837136/54364000-fb19f600-4663-11e9-9797-d8919d6a53e1.png)
![Screen Shot 2019-03-14 at 14 16 20](https://user-images.githubusercontent.com/837136/54364001-fb19f600-4663-11e9-8e27-4ad4db4ec98f.png)
![Screen Shot 2019-03-14 at 14 16 34](https://user-images.githubusercontent.com/837136/54364007-fd7c5000-4663-11e9-9d3e-4c3cea1ffb77.png)
![Screen Shot 2019-03-14 at 14 16 43](https://user-images.githubusercontent.com/837136/54364008-fd7c5000-4663-11e9-9c1f-d9f119ca2a3f.png)
![Screen Shot 2019-03-14 at 14 17 00](https://user-images.githubusercontent.com/837136/54364010-fd7c5000-4663-11e9-9edc-e55bd9c84a43.png)
![Screen Shot 2019-03-14 at 14 17 12](https://user-images.githubusercontent.com/837136/54364011-fe14e680-4663-11e9-9495-48c47a3db2b5.png)
![Screen Shot 2019-03-14 at 14 17 15](https://user-images.githubusercontent.com/837136/54364012-fe14e680-4663-11e9-949a-87b48af17828.png)
![Screen Shot 2019-03-14 at 14 17 25](https://user-images.githubusercontent.com/837136/54364014-fe14e680-4663-11e9-8868-d504412f5a89.png)

**Comment by baybal** [comment]
Ok, xlink would it be...

**Comment by edent** [comment]
Yes, I think we need to add them back in. Will work for some of the icons - but others will be too large.

**Comment by judeGibbons** [comment]
This would add 49 bytes if href is used once, and 6 more bytes for each subsequent use. I agree it should be done, probably for all of them for consistency even if href isn't used in the file. (However, ré DuckDuckGo... 😭)

**Comment by baybal** [comment]
@edent can you check if adding namespace prefix works without namespace declaration is Safari?

**Comment by edent** [comment]
Just adding the namespace does *not* work.

Best I can do is `xmlns:x="http://www.w3.org/1999/xlink"` which then lets you do `<use x:href="#c"`

So that's an extra 39 bytes for the namespace, and an extra 2 for each use. Not terrible.

**Comment by baybal** [comment]
I'll switch all uses of unnamespaced href to x:

**Comment by edent** [comment]
Thanks :-)

**Comment by edent** [comment]
Ah, there's a problem. `xmlns:x=` doesn't work when viewing the files in Linux or when converting to Android Drawable. That *requires* `xmlns:xlink=`

**Comment by baybal** [comment]
For as long as we do "web first," I see no issue there. Hardcoded namespaces are, well, hardcoded namespaces...

**Comment by edent** [comment]
I think you're probably right. Just makes it a pain when generating the Android Vectors. Oh well!

**Comment by chris-morgan** [comment]
When SVG is parsed with the HTML parser (which mostly means: when you use inline SVG), `x:href` won’t work; it does exact matching for `xlink:href`, because it’s not actually handling XML namespaces.

I extremely strongly recommend migrating back from x to xlink.
